src/GitHub.TeamFoundation.14/Home/ForkNavigationItem.cs

```diff
@@ -0,0 +1,117 @@
+using GitHub.Extensions;
+using GitHub.Logging;
+using Serilog;
+using System.Collections.Specialized;
+using GitHub.Settings;
+
+namespace GitHub.VisualStudio.TeamExplorer.Home
+{
+    [TeamExplorerNavigationItem(ForkNavigationItemId, NavigationItemPriority.Fork)]
+    [PartCreationPolicy(CreationPolicy.NonShared)]
+    public class ForkNavigationItem : TeamExplorerNavigationItemBase
+    {
+        static readonly ILogger log = LogManager.ForContext<ForkNavigationItem>();
+
+        public const string ForkNavigationItemId = "5245767A-B657-4F8E-BFEE-F04159F1DDA6";
+
+        readonly IDialogService dialogService;
+        readonly IPackageSettings packageSettings;
+        readonly IUsageTracker usageTracker;
+
+        IConnectionManager connectionManager;
+
+        [ImportingConstructor]
+        public ForkNavigationItem(IGitHubServiceProvider serviceProvider,
+            ISimpleApiClientFactory apiFactory,
+            ITeamExplorerServiceHolder holder,
+            IDialogService dialogService,
+            IPackageSettings packageSettings,
+            IUsageTracker usageTracker)
+            : base(serviceProvider, apiFactory, holder, Octicon.repo_forked)
+        {
+            this.dialogService = dialogService;
+            this.packageSettings = packageSettings;
+            this.usageTracker = usageTracker;
+
+            Text = Resources.ForkNavigationItemText;
+            ArgbColor = Colors.PurpleNavigationItem.ToInt32();
+            ConnectionManager.Connections.CollectionChanged += ConnectionsChanged;
+
+            packageSettings.PropertyChanged += (sender, args) =>
+            {
+                if (args.PropertyName == nameof(packageSettings.ForkButton))
+                {
+                    IsVisible = packageSettings.ForkButton;
+                }
+            };
+        }
+
+        IConnectionManager ConnectionManager
+        {
+            get
+            {
+                // We can't receive IConnectionManager in the constructor because Invalidate()
+                // is called from the base constructor and so we don't have chance to save it
+                // to a field.
+                if (connectionManager == null)
+                {
+                    connectionManager = ServiceProvider.GetService<IConnectionManager>();
+                }
+
+                return connectionManager;
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            base.Dispose(disposing);
+            ConnectionManager.Connections.CollectionChanged -= ConnectionsChanged;
+        }
+
+        public override async void Execute()
+        {
+            var connection = await connectionManager.GetConnection(ActiveRepo);
+
+            if (connection?.IsLoggedIn == true)
+            {
+                usageTracker.IncrementCounter(model => model.NumberOfShowRepoForkDialogClicks).Forget();
+                await dialogService.ShowForkDialog(ActiveRepo, connection);
```

Comment:
Can we expose this as a VS command? I'm trying to make sure we expose all of our commands a VS commands. See:
https://github.com/github/VisualStudio/pull/1638.
https://github.com/github/VisualStudio/commit/437f1619e117f67ff08e17d346836560f5b1f863#diff-6e8b5db6949b54276068dee9aef6d3f9

Reply:
I'll add this to the backlog for this feature...